service/routes.py

```diff
@@ -174,24 +165,69 @@ def delete_products_from_wishlist(wishlist_id):
       ),
       status.HTTP_404_NOT_FOUND
     )
+
+  product_list = wishlist.read()["products"]
+  product_ids_list = [p["id"] for p in product_list]
+  app.logger.info("Request to delete all the products from wishlist %s"\
+    % (wishlist_id))
+
   cnt = wishlist.delete_products(product_ids_list)
 
-  if cnt == len(product_ids_list):
+  if cnt != 0:
     return make_response(
       jsonify(
         data = [],
-        message = "All products are deleted"
+        message = "All products are deleted."
       ),
       status.HTTP_200_OK
     )
 
   return make_response(
     jsonify(
       data = [],
-      message = "{} products are deleted".format(cnt)
+      message = "There is no products in the wishlist, 0 products are deleted."
     ),
     status.HTTP_206_PARTIAL_CONTENT
   )
+######################################################################
+# Delete a product from a wishlist
+######################################################################
+@app.route("/wishlists/<int:wishlist_id>/products/<int:product_id>", methods=["DELETE"])
+def delete_a_product_from_wishlist(wishlist_id, product_id):
+  """
+  Delete a product from a wishlist
+  This endpoint will delete a product from a wishlist based the wishlist_id
+  and the product_id specified in the URL
+  """
+  app.logger.info("Request to delete products with id %s from wishlist %s"\
+    % (product_id,wishlist_id))
+
+  wishlist = Wishlist.find_by_id(wishlist_id)
+  if not wishlist:
+    return make_response(
+      jsonify(
+        data = [],
+        message = "Wishlist {} not found".format(wishlist_id)
+      ),
+      status.HTTP_404_NOT_FOUND
+    )
+  cnt = wishlist.delete_products([product_id])
+
+  if cnt == 0:
+    return make_response(
+      jsonify(
+        data = [],
+        message = "Product with id {} is not in this wishlist.".format(product_id)
+      ),
+      status.HTTP_206_PARTIAL_CONTENT
```

Comment:
Hmmm.... I have kind of struggled to understand what 206 really means and if it's applicable here. I know we had it before, but as far as I understand now, a 206 is returned when the server has only sent back a part of a bigger chunk of data (half of a video file, or first 100 records in the database, or something like that). So my thinking is maybe we should just stick with a 200_OK, but keep different messages for deleted product/was not there product. Thoughts?

Reply:
I agree with you, I will change them.